pkg/api/validate/vm.go

```diff
@@ -7,66 +7,97 @@ import (
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
+var supportedMasterVMSizes = []api.VMSize{
+	// General purpose
+	api.VMSizeStandardD8sV3,
+	api.VMSizeStandardD16sV3,
+	api.VMSizeStandardD32sV3,
+	// Memory optimized
+	api.VMSizeStandardE64iV3,
+	api.VMSizeStandardE64isV3,
+	api.VMSizeStandardE80isV4,
+	api.VMSizeStandardE80idsV4,
+	api.VMSizeStandardE104iV5,
+	api.VMSizeStandardE104isV5,
+	api.VMSizeStandardE104idV5,
+	api.VMSizeStandardE104idsV5,
+	// Compute optimized
+	api.VMSizeStandardF72sV2,
+	// Memory and storage optimized
+	api.VMSizeStandardGS5,
+	api.VMSizeStandardG5,
+	// Memory optimized
+	api.VMSizeStandardM128ms,
+}
```

Comment:
Why not use a map[api.VMSize]struct{} or map[api.VMSize]bool instead of doing a lookup with a loop?